### PR TITLE
smn charts

### PR DESCRIPTION
**Pull request by docwarren**

### CODE REVIEW

**Comment by docwarren** [comment]
Xiao - are you going to merge this?

**Comment by xiao-chen-xc** [comment]
> Xiao - are you going to merge this?

Yes, will try to find some time this week. Sorry about the delay.

File: charts/draw.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    lines_height = conf["height"] + conf["padding"]
+    bars_height = conf["height"] + conf["padding"]
+    return histo_height + lines_height + bars_height
+
+
+def svg_file(conf, sample):
+    return "%s/smn_%s.svg" % (conf["output_dir"], sample)
+
+
+def png_file(conf):
```
Comment (xiao-chen-xc) [inline charts/draw.py]: This should probably be deleted since there is no png output?

File: charts/histogram.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    x_axis = get_x_axis(pop_data, width, padding)
+    y_axis = get_y_axis(pop_data, height, padding)
+
+    if fmt == "svg":
+        return svg_histo.get_svg(pop_data, sample_data, config, x_axis, y_axis, col=col, header=header[col])
+    elif fmt == "pdf":
+        return pdf_histo.get_pdf(pop_data, sample_data, x_axis, y_axis, col=col, header=header[col])
+
+
+def add_to_map(data_map, key):
```
Comment (xiao-chen-xc) [inline charts/histogram.py]: This is not used and is also redundant with the same function in data_utils

File: smn_charts.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+if __name__ == "__main__":
+    config = {
+        "set_width": 800,
+        "set_height": 200,
+        "set_padding": 40,
+        "width": 800,
+        "height": 200,
+        "padding": 40,
+        "pop_file": "/Users/awarren/Documents/Illumina/XiaoSMN/SMN_1kGP_results.tsv",
```
Comment (xiao-chen-xc) [inline smn_charts.py]: Could we remove these local file names?

File: charts/draw.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            sample_col_map = {sample: sample_data[sample][col]}
+            hist = histo.get_histogram(pop_col, sample_col_map, conf, col, "pdf")
+            add_space()
+            elements.append(pdf.add_chart_to_page(drawing, hist))
+            idx += 1
+
+        drawing = Drawing(conf["width"], conf["height"], vAlign="TOP")
+        col_map = util.get_key_map(conf["line_charts"]["columns"], sample_data[sample])
+        cn = round(sample_data[sample]["Full_length_CN_raw"]) / 2
+        col_map["SMN2_CN_raw"] = [(x[0], (cn - (x[1] - cn))) for x in col_map["SMN1_CN_raw"]]
```
Comment (xiao-chen-xc) [inline charts/draw.py]:
```suggestion
        cn = round(sample_data[sample]["Full_length_CN_raw"])
        col_map["SMN2_CN_raw"] = [(x[0], cn - x[1]) for x in col_map["SMN1_CN_raw"]]
```

Comment (xiao-chen-xc) [inline charts/draw.py]:
```suggestion
        full_length_cn = round(sample_data[sample]["Full_length_CN_raw"])
        col_map["SMN2_CN_raw"] = [(x[0], full_length_cn - x[1]) for x in col_map["SMN1_CN_raw"]]
```

File: charts/draw.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+            pop_col = util.get_pop_column(pop_data, col)
+            sample_col_map = {sample: sample_data[sample][col]}
+            hist = histo.get_histogram(pop_col, sample_col_map, conf, col, "svg")
+            idx += 1
+            page = svg.add_chart_to_page(page, hist)
+
+        conf["index"] = idx
+        col_map = util.get_key_map(conf["line_charts"]["columns"], sample_data[sample])
+        cn = round(sample_data[sample]["Full_length_CN_raw"]) / 2
+        col_map["SMN2_CN_raw"] = [(x[0], (cn - (x[1] - cn))) for x in col_map["SMN1_CN_raw"]]
```
Comment (xiao-chen-xc) [inline charts/draw.py]: Same as above

File: smn_charts.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+from charts.draw import write_svg, write_pdf
+
+
+def get_args():
+    parser = argparse.ArgumentParser(description='Convenient pdf and svg charts for smn_caller output.')
+
+    parser.add_argument('-p', '--pop-data',
+                        dest="pop_data",
+                        help='smn_caller.py output: Population tsv file',
+                        required=True)
```
Comment (xiao-chen-xc) [inline smn_charts.py]: Can we include the SMN_1kGP_results.tsv with the code, e.g. put it in charts/data/, and make this parameter optional? The distribution should be stable and no need to require user input, unless the user wants to examine their own distribution.

File: smn_charts.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    args = get_args()
+    conf["pop_file"] = args.pop_data
+    conf["sample_file"] = args.sample_data
+    conf["output_dir"] = args.out_dir
+
+    pop_data = util.read_pop_data(conf["pop_file"])
+    sample_data = util.read_sample_data(conf["sample_file"])
+
+    write_svg(conf, pop_data, sample_data)
+    write_pdf(conf, pop_data, sample_data)
```
Comment (xiao-chen-xc) [inline smn_charts.py]: I think that we probably don't need to produce two output files per sample. Could we ask the user to specify if they want pdf or svg (maybe set pdf as the default)?

**Review by xiao-chen-xc — COMMENTED** [review]
@docwarren I have made a few comments. Sorry about the delay.

File: smn_charts.py [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+                        help='Draw to svg instead of pdf',
+                        required=False)
+
+    args = parser.parse_args()
+    return args
+
+
+def main(conf):
+    args = get_args()
+    conf["pop_file"] = args.pop_data
```
Comment (xiao-chen-xc) [inline smn_charts.py]:
```suggestion
    if args.pop_data is not None:
        conf["pop_file"] = args.pop_data
```

File: smn_charts.py [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+if __name__ == "__main__":
+    config = {
+        "set_width": 800,
+        "set_height": 200,
+        "set_padding": 40,
+        "width": 800,
+        "height": 200,
+        "padding": 40,
+        "pop_file": "./charts/data/SMN_1kGP_results.tsv",
+        "sample_file": "./charts/data/example_smn.json",
```
Comment (xiao-chen-xc) [inline smn_charts.py]: I don't think relative paths would work here. Could you change to use something like `os.path.dirname(__file__)` instead?

**Review by xiao-chen-xc — COMMENTED** [review]
@docwarren Thanks! Just a few very minor comments in `smn_charts.py`.

File: smn_charts.py (hunk 8 above)
Comment (xiao-chen-xc) [reply smn_charts.py]: @docwarren how about this one? pop_data is not required so take the value only when it's provided.

**Comment by xiao-chen-xc** [comment]
Looks good. Thanks!

**Pull request merged**